crates/agent/src/publications.rs

```diff
@@ -196,9 +201,54 @@ impl PublishHandler {
             .await;
         }
 
-        let live_spec_ids: Vec<_> = spec_rows.iter().map(|row| row.live_spec_id).collect();
+        let expanded_rows = specs::expanded_specifications(row.user_id, &spec_rows, txn).await?;
+        tracing::debug!(specs = %expanded_rows.len(), "resolved expanded specifications");
+
+        let errors = draft::extend_catalog(
+            &mut draft_catalog,
+            expanded_rows
+                .iter()
+                .map(|r| (r.live_type, r.catalog_name.as_str(), r.live_spec.0.as_ref())),
+        );
+        if !errors.is_empty() {
+            anyhow::bail!("unexpected errors from expanded specs: {errors:?}");
+        }
+
+        // It's important that we prune unbound collection specs _after_ the catalog is expanded,
+        // and before checking resource quotas
+        let pruned_collections =
+            specs::prune_unbound_collections(&mut draft_catalog, &mut spec_rows);
+        // `resolve_specifications` will have created `live_specs` for each of the `draft_specs` in the publication.
+        // We need to delete any of those that correspond
```

Comment:
Unfinished comment?

Reply:
this was superceded by the new approach / fixed